app/src/main/java/org/p2p/wallet/newsend/ui/search/NewSearchPresenter.kt

```diff
@@ -30,18 +31,25 @@ class NewSearchPresenter(
     private var lastQuery: String? = null
     private var searchJob: Job? = null
     private var lastResult: List<SearchResult> = emptyList()
+    private var recentRecipients: List<SearchResult> = emptyList()
 
     override fun loadInitialData() {
-        val data = lastResult.takeIf { lastResult.isNotEmpty() } ?: usernames
-        if (data.isNullOrEmpty()) {
-            // TODO PWN-6077 check latest recipients and show if exists
-            view?.showEmptyState(isEmpty = true)
-        } else {
-            val searchedItem = data.first()
-            val value = (searchedItem as? SearchResult.UsernameFound)?.username
-                ?: searchedItem.addressState.address
-            view?.showSearchValue(lastQuery ?: value)
-            setResult(data)
+        launch {
+            val finalSearchResult = lastResult.takeIf { lastResult.isNotEmpty() } ?: usernames
+            if (finalSearchResult.isNullOrEmpty()) {
+                recentRecipients = userInteractor.getRecipients()
+                if (recentRecipients.isEmpty()) {
+                    view?.showEmptyState(isEmpty = true)
+                } else {
+                    setResult(recentRecipients, R.string.search_recently)
+                }
+            } else {
+                val searchedItem = finalSearchResult.first()
+                val value = (searchedItem as? SearchResult.UsernameFound)?.username
+                    ?: searchedItem.addressState.address
+                view?.showSearchValue(lastQuery ?: value)
+                setResult(finalSearchResult)
+            }
```

Comment:
this function is getting complex too. 
add todo to make it more optimized please

Reply:
Done